### PR TITLE
Change from Virtus.model to Virtus.value_object.

### PR DESCRIPTION
**Pull request by knappe**

### CODE REVIEW

**Comment by knappe** [comment]
@ryanwi @dmvt Is this the only change needed?  I'm a little in the dark on this change.

The specs pass and I also was able to send a test message using the gem on for a project that depends on the gem and those specs also all pass for that project.

**Comment by ryanwi** [comment]
Right, in theory. From what I had scanned how the apps publish messages, it should be a seamless transition. Value objects are immutable, so the issue would be if any producers attempt to change properties on the messages. So, scanning v3's producing to ensure that isn't done is the other part of this. We can look at smaller producers first as well as a test.

**Comment by knappe** [comment]
Gotcha, I'll take a look at v3.

**Comment by ryanwi** [comment]
Of particular note is @kyleries work on changeset translation, does that depend on using Virtus model?

https://github.com/emque/emque-producing/commit/7883b5c4de4a1ceae9be4aa662ee746f2200313f

**Comment by knappe** [comment]
The changeset tests are all passing, so I'd _think_ those are good.

I took a look at v3 and didn't see any messages being modified, so I think we're clear here as well.

**Comment by dmvt** [comment]
:shipit: with a README update

**Comment by knappe** [comment]
@ryanwi @dmvt Any strong feelings on how the changelog is formatted?

**Comment by knappe** [comment]
@dmvt What readme update would you like?  A note that we're using `Virtus.value_object` and not `Virtus.model`?  I didn't see anything specific on the Readme.

**Comment by knappe** [comment]
@dmvt Sorry, I missed that the README was using the old attribute definition.  Updated!

**Comment by ryanwi** [comment]
Looks good, is there an issue or description of the reason? (I know it's for immutability and improving compatibility with consuming, but wondering if that'll be clear as we look back at the changelog)

**Comment by knappe** [comment]
@ryanwi No, there isn't an issue for this change.  We could create one and link it back to this PR to ensure we have the history.

**Comment by knappe** [comment]
Fixes https://github.com/emque/emque-producing/issues/42.

File: spec/producing/message/changes_payload_generator_spec.rb [hunk 1]
```diff
@@ -3,11 +3,13 @@
 require "emque/producing/message/message_with_changeset"
 
 class FakeModel
-  include Virtus.model
+  include Virtus.value_object
 
-  attribute :foo, Integer
-  attribute :bar, Integer
-  attribute :baz, Integer
+  values do
```
Comment (ryanwi) [inline spec/producing/message/changes_payload_generator_spec.rb]: Do all messages have to be changed to use the `values do` block? that could be something to work on in all of the producers

**Comment by ryanwi** [comment]
:shipit:

**Pull request merged**